File: toolkit_algoritms/week3/c_code/car_fueling.cpp
#include <iostream>

int MinRefills(int x[], int n, int L){
	int numRefills = 0;
	int currentRefill = 0;

	while(currentRefill <= n){
		int lastRefill = currentRefill;

		while(currentRefill <= n && ((x[currentRefill + 1] - x[lastRefill]) <= L)){
			currentRefill++;
		}

		if(currentRefill == lastRefill){

			return -1;
		}
		if(currentRefill <= n){
		numRefills++;}
	}
	return numRefills;

}

int main(){


	return 0;
}

File: toolkit_algoritms/week3/python_code/car_fueling.py


def MinRefills(x, n, L):
    numRefills = 0
    currentRefills = 0
    while(currentRefills <= n):
        lastRefills = currentRefills


        while(currentRefills <= n and (x[currentRefills + 1] - x[lastRefills]) <= L):
            currentRefills += 1

        if(currentRefills == lastRefills):
            return -1

        if(currentRefills <= n):
            numRefills += 1

    return numRefills




if __name__ == '__main__':
    pass
